hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientSmartInvocationServiceImpl.java

```diff
@@ -70,10 +69,29 @@ public void invokeOnTarget(ClientInvocation invocation, Address target)
         if (!isMember(target)) {
             throw new IOException("Target :  " + target + " is not member. ");
         }
-        final Connection connection = connectionManager.getOrConnect(target, authenticator);
+        final Connection connection = getConnection(target);
         invokeOnConnection(invocation, (ClientConnection) connection);
     }
 
+    private Connection getConnection(Address target) throws IOException {
+        ensureOwnerConnection();
+        return connectionManager.getOrConnect(target, authenticator);
+    }
+
+    private void ensureOwnerConnection() throws IOException {
```

Comment:
The name is a obscure. What does it mean 'ensureOwnerConnection' ?? Should that no be e.g. 'ensureOwnerConnectionAvailable' ?